tests/FSharp.Compiler.ComponentTests/EmittedIL/Literals.fs

```diff
@@ -176,34 +176,24 @@ let [<Literal>] x = System.Int32.MaxValue + 1
         }
 
     [<Fact>]
-    let ``Compilation fails when using decimal arithmetic in literal``() =
+    let ``Arithmetic can be used for constructing decimal literals``() =
         FSharp """
 module LiteralArithmetic
 
-let [<Literal>] x = 1m + 1m
+[<Literal>]
+let x = 1m + 2m
         """
         |> withLangVersion80
         |> compile
-        |> shouldFail
-        |> withResults [
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 21
-                        EndLine = 4
-                        EndColumn = 23 }
-              Message = "This is not a valid constant expression or custom attribute value" }
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 26
-                        EndLine = 4
-                        EndColumn = 28 }
-              Message = "This is not a valid constant expression or custom attribute value" }
-            { Error = Error 267
-              Range = { StartLine = 4
-                        StartColumn = 21
-                        EndLine = 4
-                        EndColumn = 28 }
-              Message = "This is not a valid constant expression or custom attribute value" }
+        |> shouldSucceed
+        |> verifyIL [
+            """.field public static initonly valuetype [runtime]System.Decimal x"""
```

Comment:
I think addition codegen might be required - initialization of the constant in the cctor

Reply:
Could you also please add an explicit baseline that a cctor (static one) is codegened for the type we emit field for, and manually verify that C# is generating roughly the same IL (if you didn't do so already)? Thanks!